ui/src/app/machines/views/MachineList/MachineList.js

```diff
@@ -51,17 +70,24 @@ const DnsForm = () => {
           </div>
         )}
         {machinesLoaded && (
-          <ul className="p-list">
-            {machines.map(machine => (
-              <li className="p-list__item" key={machine.fqdn}>
-                {machine.fqdn}
-              </li>
-            ))}
-          </ul>
+          <MainTable
+            className="p-table-expanding--light"
+            defaultSort="status"
+            defaultSortDirection="ascending"
+            headers={[
+              {
+                content: "FQDN",
+                sortKey: "name"
+              }
+            ]}
+            paginate={150}
+            rows={generateRows(machines)}
+            sortable
+          />
         )}
       </Col>
     </Row>
   );
 };
 
-export default DnsForm;
```

Comment:
I feel like `DnsForm` captures the _spirit_ of this component a bit better, though `MachineList` is probably more conceptually honest.